### DISCUSSION TITLE
CentOS8 distro installed from MS Store doesn't start giving "Can't find file"

### DISCUSSION THREAD

**Discussion by PavelSosin-320** [microsoft/WSL#5794] [General]
# Environment:
Microsoft Windows [Version 10.0.20190.1000]

```none
Windows build number: [run `[Environment]::OSVersion` for powershell, or `ver` for cmd]
CentOS8
WSL 2
```

# Steps to reproduce
1. Install CentOS8 distro
2. Launch using any method - everything is the same.

No logs - because of no access
.wslconfig doesn't exist in %USERPROFILE/AppData/Lcal/Tmp
Nothing related to [wslconfig](https://docs.microsoft.com/en-us/windows/wsl/wsl-config) exists!
Some swap files exist, maybe related to other working distros like CentOS7: like {99b21a4a-725d-4a99-b51b-d5f121efb31c}.vhdx 64Mb - swapfile ?
#  Expected behavior
Distro starts like others: Ubuntu 20.04, CentOS7
# Actual behavior
Fails to start with error "The system cannot find the file specified."
[process exited with code 4294967295]

> when running wsl -d CentOS8 -u root
The system cannot find the file specified.
CentOS8 distro is unavailable by any kind of access!
P.S. I suppose CentOS 8 needs its own kernel to support numerous RedHat applications. CentOS kernel is available in RedHat site

**Comment by licanhua** [comment]
Could you please check if %SystemRoot%\system32\lxss\tools\kernel exists or not? If not, please follow https://aka.ms/wsl2kernel to install kernel.

**Comment by PavelSosin-320** [comment]
Finally, I did it using RootFS image from CentOS 8.1 GitHub. But then I faced 2 major problems:
1. The documentation for this project exists in Japanese only.
2. CentOS 8 is essentially systemd based and without systemd support, nothing can be used except bash shell. The main CentOS8 asset Podman can't run without systemd - it can't be started manually like docker daemon.

**Comment by WSLUser** [comment]
> CentOS 8 is essentially systemd based and without systemd support, nothing can be used except bash shell. The main CentOS8 asset Podman can't run without systemd - it can't be started manually like docker daemon.

 See https://github.com/microsoft/WSL/issues/994

**Reply by PavelSosin-320** [reply]
Ubuntu team has solved systemd issue in WSL a long time ago and I use Podman from Ubuntu-20.04 distro for already a long time. Unlike Docker, Podman doesn't explicitly require systemd because it is not a server. The real problem is that poor WSL distros management in Microsoft Store and Linux Kernel updates make any WSL distro related issues to be a long-lasting problem.  CentOS8 distro doesn't copy its RootFS to the local machine during installation at all. This is a simple truth.
